src/linalg/basic/matrix.rs

```diff
@@ -544,15 +627,74 @@ mod tests {
     use approx::relative_eq;
 
     #[test]
-    fn test_display() {
+    fn test_instantiate_from_2d() {
         let x = DenseMatrix::from_2d_array(&[&[1., 2., 3.], &[4., 5., 6.], &[7., 8., 9.]]);
+        assert!(x.is_ok());
+    }
+    #[test]
+    fn test_instantiate_from_2d_empty() {
+        let input: &[&[f64]] = &[&[]];
+        let x = DenseMatrix::from_2d_array(input);
+        assert!(x.is_err());
+    }
+    #[test]
+    fn test_instantiate_from_2d_empty2() {
+        let input: &[&[f64]] = &[&[], &[]];
+        let x = DenseMatrix::from_2d_array(input);
+        assert!(x.is_err());
+    }
+    #[test]
+    fn test_instantiate_ok_view1() {
```

Comment:
These tests are failing:
```
test linalg::basic::matrix::tests::test_instantiate_err_view1 ... FAILED
test linalg::basic::matrix::tests::test_instantiate_err_view2 ... FAILED
test linalg::basic::matrix::tests::test_instantiate_err_view3 ... FAILED
test linalg::basic::matrix::tests::test_instantiate_from_2d ... ok
test linalg::basic::matrix::tests::test_instantiate_from_2d_empty ... FAILED
test linalg::basic::matrix::tests::test_instantiate_from_2d_empty2 ... FAILED
```

Reply:
```
thread 'linalg::basic::matrix::tests::test_instantiate_err_view1' panicked at 'The window end is outside of the matrix range', src/linalg/basic/matrix.rs:61:5
stack backtrace:
```

Reply:
They pass in a release mode. The assertions can be guarded with #[cfg debug]. So, only when the debug flag is active will the assertions kick-in.

Reply:
tests should pass by simply running `cargo test` from the command line, if they don't we should change something or in the tests configuration or in the debug assertions to make them pass.
I would remove the debug assertions as they make the code less readable, assertions should be present only in test functions.

Reply:
I'll go ahead and update the branch so that the tests pass in non-optimized mode as well as release.

Reply:
The tests now pass in in non-optimized mode.